ipatests/test_integration/test_replica_promotion.py

```diff
@@ -1013,13 +1014,19 @@ def test_hidden_replica_install(self):
         self._check_dnsrecords([self.master], [self.replicas[0]])
         self._check_config([self.master], [self.replicas[0]])
 
-    @pytest.mark.xfail(
-        reason='https://pagure.io/freeipa/issue/8582', strict=True
-    )
     def test_ipahealthcheck_hidden_replica(self):
         """Ensure that ipa-healthcheck runs successfully on all members
         of an IPA cluster that includes a hidden replica.
         """
+        os_version = (osinfo.id, osinfo.version_number[0])
```

Comment:
Here you are getting the os version of the controller, not of the master/replica.
We already have [tasks.get_platform()](https://github.com/freeipa/freeipa/blob/50241b36afe9019543dda0f8387ff25e6c672426/ipatests/pytest_ipa/integration/tasks.py#L2435), you can use something similar to retrieve the information you need for a remote host.